File: 01 - FUNDAMENTOS DE PROGRAMAÇÃO/C/Exercicio 8.c
/* 
8 - Elabore um algoritmo para calcular e imprimir o valor de S, sendo S = -1/2 + 2/4 - 3/6 + 4/8
-5/10 +, ..., +10/20 
*/

#include<stdio.h>

int main ()
{
    float s=0;
    int i, den=2;

    for (i=1; i<=10; i++){
        if (i%2 == 0)
            s=s+(i/(float)den);
        else
            s=s+((-i)/(float)den);
        den += 2;
    }
    printf ("\n A soma eh de: %f\n\n",s);
    return 0;
}

File: 01 - FUNDAMENTOS DE PROGRAMAÇÃO/Python/exercicio8.py
"""
8 - Elabore um algoritmo para calcular e imprimir o valor de S, sendo S = -1/2 + 2/4 - 3/6 + 4/8 -5/10 +, ..., +10/20
"""

s=0
numerador=1
denominador=2

for numerador in range(1, 10+1):
  if(numerador%2==0):
    s += (float)(numerador/denominador)
  else:
    s += (float)((-numerador)/denominador)
  denominador += 2

print(s)